packages/azure-http-specs/specs/azure/client-generator-core/api-version/main.tsp

```diff
@@ -0,0 +1,36 @@
+import "@typespec/http";
+import "@typespec/spector";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using Azure.ClientGenerator.Core;
+using Spector;
+
+@doc("Test for api-version decorator.")
+@scenarioService("/azure/client-generator-core/api-version")
+namespace _Specs_.Azure.ClientGenerator.Core.ApiVersion;
```

Comment:
I assume if we are testing azure api-version, the service itself should be versioned (instead of just value "1234" in mockapi)?
E.g. https://github.com/Azure/typespec-azure/blob/020ca109d6fad08b8de8fcd6ea264ad6ef43d277/packages/azure-http-specs/specs/azure/core/basic/main.tsp#L16-L31

Also, unsure how client works when 3 different scheme of api-version appears in a same client. Need to test.

Reply:
Separated out the scenarios into different specs. Also, ci seems to be fixed now